avocado/utils/cpu.py

```diff
@@ -195,3 +197,59 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpuidle_state():
+    """
+    Get current cpu idle values
+
+    :return: Dict of cpuidle states values for all cpus
+    :rtype: Dict of dicts
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idlestate = {}
+    for cpu in cpus_list:
+        cpu_idlestate[cpu] = {}
+        for state_no in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (cpu, state_no)
+            try:
+                cpu_idlestate[cpu][state_no] = int(open(state_file).read())
+            except IOError as err:
+                logging.warning("Failed to read idle state on cpu %s "
+                                "for state %s:\n%s", cpu, state_no, err)
+    return cpu_idlestate
+
+
+def set_cpuidle_state(state_number="all", disable=1, oldstate=None):
+    """
+    Set/Reset cpu idle states for all cpus
+
+    :param state_number: cpuidle state number, default: `all` all states
+    :param disable: whether to disable/enable given cpu idle state, default: 1
+    :param oldstate: old cpuidle state value, output of `get_cpuidle_state()`
+    """
+    cpus_list = cpu_online_list()
+    if not oldstate:
+        states = []
+        if state_number == 'all':
+            states = range(0, len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+        else:
+            states.append(state_number)
+        for cpu in cpus_list:
+            for state_no in states:
+                state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (cpu, state_no)
+                try:
+                    open(state_file, "w").write(str(disable))
+                except IOError as err:
+                    logging.warning("Failed to set idle state on cpu %s "
+                                    "for state %s:\n%s", cpu, state_no, err)
+    else:
+        for cpu in cpus_list:
```

Comment:
well this can mask/fail in case the oldstate is not coherent with the current situation. I'd suggest using oldstate.items() to get all cpus from the state.